File: thematic_tutorials/bhax_textbook_IgyNeveldaProgramozod/src/Liskov/Family.cpp
#include <iostream>
#include <string>


class Szulo
{
	public:
        void message()
        {
            std::cout << "Uzenet" << std::endl;
        }
};

class Gyerek : public Szulo
{
	public:
        void message2()
        {
            std::cout << "Gyerek uzenet" << std::endl;
        }
};


int main()
{
	Szulo* szulo = new Szulo();
	Szulo* gyerek = new Gyerek();

    szulo->message();   //Lefut
	gyerek->message2(); //Nem fut le
	delete szulo;
	delete gyerek;

}

File: thematic_tutorials/bhax_textbook_IgyNeveldaProgramozod/src/Liskov/Family.java
class Szulo{

    void message() {
        System.out.println("Szülő üzenete.");
    }
}

class Gyerek extends Szulo{

    void message2() {
        System.out.println("Gyerek üzenete");
    }
}

class Family{
    public static void main(String[] args) {

        Szulo szulo = new Szulo();
        Szulo gyerek = new Gyerek();

        szulo.message();
        gyerek.message2();      

    }
}